src/Umbraco.Web/Strategies/RelateOnTrashHandler.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Linq;
+using Umbraco.Core;
+using Umbraco.Core.Auditing;
+using Umbraco.Core.Events;
+using Umbraco.Core.Models;
+using Umbraco.Core.Services;
+
+namespace Umbraco.Web.Strategies
+{
+    public sealed class RelateOnTrashHandler : ApplicationEventHandler
+    {
+        protected override void ApplicationStarting(UmbracoApplicationBase umbracoApplication, ApplicationContext applicationContext)
```

Comment:
Yes, that's why i moved the other one to Core. There's a reason things are in Core vs Web, if there's no dependency on Web and it's a data driven thing then it should be in Core. Umbraco.Web should really only contain any logic for running the web app in a web context. In some cases there's some event handlers that need to exist there because of our legacy assembly references though.